byterun/win32.c

```diff
@@ -914,3 +916,22 @@ void caml_restore_win32_terminal(void)
   if (startup_codepage != 0)
     SetConsoleOutputCP(startup_codepage);
 }
+
+static LARGE_INTEGER frequency;
+
+void caml_init_os_params(void)
+{
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  caml_sys_pagesize = si.dwPageSize;
+  QueryPerformanceFrequency(&frequency);
+}
+
+int64_t caml_time_counter(void)
+{
+  LARGE_INTEGER now;
+  /* Windows 2000 is no longer supported, so this function always succeeds */
+  QueryPerformanceCounter(&now);
+  now.QuadPart *= 1000000000L;
+  return (now.QuadPart * 1000000000L) / frequency.QuadPart;
```

Comment:
This doesn't look right. First, it does the sec->nanosecs conversion twice. Second, even if only done once, I think the multiplication will overflow after less than a day.

Reply:
Yeah, I seem to have turned my brain off there.